lib/scryptenc/scryptenc.c

```diff
@@ -134,7 +134,7 @@ pickparams(size_t maxmem, double maxmemfrac, double maxtime,
 		*p = 1;
 		maxN = opslimit / (*r * 4);
 		for (*logN = 1; *logN < 63; *logN += 1) {
-			checkN = (uint64_t)(1) << *logN;
+			checkN = (double)((uint64_t)(1) << *logN);
```

Comment:
Why are we casting an integer to a floating-point type again?

Reply:
`maxN` is a double, because `opslimit` is a double, because `opps` and `maxtime` are doubles.

That said, I now think it makes more sense to make `checkN` a uint64_t, then have
```
if ((double)checkN > maxN / 2)
```

(Another option would be to change `opslimit` to an integer type, but I assume that you deliberately didn't do that.)